sycl/include/CL/sycl/handler.hpp

```diff
@@ -128,6 +128,30 @@ struct check_fn_signature<F, RetT(Args...)> {
 
 __SYCL_EXPORT device getDeviceFromHandler(handler &);
 
+template <int Dims> void throwIfNotIntRange(const range<Dims> &Range) {
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__)
+  static constexpr size_t Limit = static_cast<size_t>(INT_MAX);
+  for (size_t Dim = 0; Dim < Dims; ++Dim)
+    if (Range[Dim] > Limit)
+      throw runtime_error("Provided range is out of integer limits. Suggest "
+                          "disabling `id-queries-fit-in-int32' optimizations "
+                          "flag.",
+                          PI_INVALID_VALUE);
+#endif
+}
+
+template <int Dims> void throwIfNotIntOffset(const id<Dims> &Id) {
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__)
+  static constexpr size_t Limit = static_cast<size_t>(INT_MAX);
+  for (size_t Dim = 0; Dim < Dims; ++Dim)
+    if (Id[Dim] > Limit)
+      throw runtime_error("Provided offset is out of integer limits. Suggest "
+                          "disabling `id-queries-fit-in-int32' optimizations "
+                          "flag.",
+                          PI_INVALID_VALUE);
+#endif
+}
+
```

Comment:
These functions are identical, except for id vs range. I wonder if it is possible to make a single function that accepts either of these (and maybe adjust error message with `std::is_same`).

Reply:
Right. Both: range and id are derived from the base class detail::array. Perhaps, one function with 'detail::array' operand can do the work.